test_coding_rules_test.sh

```diff
@@ -29,6 +29,15 @@
 
 echo "Testing 'test_coding_rules'."
 
+# Directory where this script resides.
+
+srcdir=$(dirname "$(readlink --canonicalize "$0")")
+
+# Cannot recursively check script on path determined at runtime, so
+# a directive like 'source="$srcdir"' doesn't work.
```

Comment:
Could we do `olddir=$(pwd); cd $srcdir; . ./set_toolchain.sh; cd $olddir`?

But to be honest I'm not sure it's worth it, we don't need to check `set_toolchain.sh` here, as we already check it independently anyhow.